### DISCUSSION TITLE
Log wrapper

### DISCUSSION THREAD

**Discussion by bailantaotao** [redis/go-redis#3015] [Ideas]
Hi, I used the Redis Pub/Sub to serve each HTTP request channel, but I can't figure out which IP is getting the `channel is full` error. Because log is global in Go-Redis. 

I propose an idea where users can pass a `log` ChannelOption in `PubSub.Channel` so they can detect which one had an error occur. 

Please~~

**Comment by ycdesu** [comment]
It's really a useful feature. Hope the pull request can be reviewed

**Comment by ndyakov** [comment]
Let's spend some time on designing a better logging, we are open to proposals to decide which interface the logger should follow.
